docs/markdoc-component-documentation.md

```diff
@@ -0,0 +1,263 @@
+
+`{% button link="https://dog.ceo/" %} Click Me! {% /button %}`
+
+{% button link="https://dog.ceo/" %} Click Me! {% /button %}
+
+### Primary
+
+`{% button link="https://dog.ceo/" style="primary" %} Click Me! {% /button %}`
+
+{% button link="https://dog.ceo/" style="primary" %} Click Me! {% /button %}
+
+### Secondary
+
+`{% button link="https://dog.ceo/" style="secondary" %} Click Me! {% /button %}`
+
+{% button link="https://dog.ceo/" style="secondary" %} Click Me! {% /button %}
+
+### Blurple
+
+` {% button link="https://dog.ceo/" style="blurple" %} Click Me! {% /button %} `
+
+{% button link="https://dog.ceo/" style="blurple" %} Click Me! {% /button %}
+
+### Button Container
+The main use case I've seen for buttons in the docs is basically setting up multiple styling links in a row. Doing this requires putting buttons into a `ButtonContainer`
+
+{% buttonContainer %} {% button link="https://dog.ceo/" style="primary" %} Click Me! {% /button %} {% button link="https://dog.ceo/" style="secondary" %} Click Me! {% /button %} {% button link="https://dog.ceo/" style="blurple" %} Click Me! {% /button %} {% /buttonContainer %}
+
+## Crosses and Checks 
+You can invoke a cross with this tag `{% cross /}%` and it looks like this {% cross /%}. You can invoke checks with this tag `{% check /}%` and it looks like this {% check /%}.
+
+Crosses and checks can also be used in lists:
+- {% check /%} Completed task
+- {% cross /%} Incomplete task
+- {% cross /%} Super Incomplete task 
+
+You can also put crosses and checks into headers 
+
+### This is a header with a check {% check /%}
+Which is pretty neat.
+
+### This is a header with a cross {% cross /%}
+Which is also pretty neat.
+
+## Images
+We've got a tag that handles making sure images are pretty, optimized, and accessible.
+
+`{% image src="/images/some-image.png" width=300 height=200 alt="Text for screenreaders. Usually you want to describe things here." /%}`
+
+<!-- {% image src="/images/concepts/assets/asset-activity-observation.png" width=300 height=200 alt="Text for screenreaders. Usually you want to describe things here." /%} -->
+
+{% image src="/images/concepts/assets/asset-activity-observation.png" width=1758 height=1146 alt="Text go here" /%}
+
+
+## Badges
+
+`{% badge text="Badgey the Badger" /%}` lets you put a custom badge onto the page like this. {% badge text="Badgey the Badger" /%}
+
+### API Status Badges
+We've also got a bunch of badges that you can use to indicate the level of support an API or feature has.
+
+There are three types of badges:
+
+- `{% experimental /%}` {% experimental /%}
+- `{% deprecated /%}` {% deprecated /%}
+- `{% legacy /%}` {% legacy /%}
+
+## Code Reference Links
+
+Code reference links let you point to a specific file in the codebase. They're useful for linking to examples or reference implementations.
+
+`{% codeReferenceLink filePath="examples/deploy_ecs" /%}`
+
+{% codeReferenceLink filePath="examples/deploy_ecs" /%}
+
+## Reference Tables
```

Comment:
Looking at this, I wonder if it's best to scrap this component and just focus on styling improvements + flexibility for regular ol' tables. Being able to use Markdown in tables like these is huge.

I made this component awhile back because I needed control that Markdown tables didn't provide (ex: setting custom widths, text alignment, etc) and I wanted to avoid a ton of table HTML. Notice how I've since avoided using it outside of a few pages 😅 

We can discuss more if needed, but basically I'd like tables that:

- I can set custom widths for columns
- Align text at the top of cells, whether it's a paragraph or list. Lists are really only an issue when they're the first thing in a cell, like [here](https://docs.dagster.io/dagster-plus/managing-deployments/branch-deployments#setting-up-branch-deployments)